packages/api/src/resolvers/user/index.ts

```diff
@@ -355,3 +361,47 @@ export const signupResolver: ResolverFn<
     return { errorCodes: [SignupErrorCode.Unknown] }
   }
 }
+
+export const deleteAccountResolver = authorized<
+  DeleteAccountSuccess,
+  DeleteAccountError,
+  MutationDeleteAccountArgs
+>(async (_, { userID }, { models, claims, log, authTrx, pubsub }) => {
+  const user = await models.user.get(userID)
+
+  if (!user) {
+    return {
+      errorCodes: [DeleteAccountErrorCode.UserNotFound],
+    }
+  }
+
+  if (user.id !== claims.uid) {
+    return {
+      errorCodes: [DeleteAccountErrorCode.Unauthorized],
+    }
+  }
+
+  const deleteUserResult = await authTrx((tx) =>
+    models.user.deleteUser(claims.uid, tx)
+  )
+
+  // delete this user's pages in elastic
+  await deletePagesByParam({ userId: userID }, { uid: userID, pubsub })
```

Comment:
hey @satindar i added deleting this user's pages in elastic

Reply:
> hey @satindar i added deleting this user's pages in elastic

Nice! Do you mind deploying this to demo? I can do some manual testing on iOS after that.

Reply:
sorry for the delay. deployed to demo! @satindar